kiwipy/rmq/communicator.py

```diff
@@ -570,6 +570,7 @@ def stop_loop():
 
         # The stop will end up setting self._communicator_thread to None
         self._loop.add_callback(stop_loop)
+        ioloop.IOLoop.clear_current()
```

Comment:
From what I can see, this is not correct.

The point in these communicators is that there can be multiple running on seperate threads.  Here `clear_current()` is called from the same thread that called `stop()` which may be the master thread.  The problem is that `clear_current` actually clears the loop of the current thread i.e. the master thread.  This is not the intended behaviour.  Calling `start` and `stop` here should in no way affect the loop of any other thread other than the communication thread created in `start`.

You could try putting `ioloop.IOLoop.clear_current()` on line 544 i.e. inside the coroutine launched on the child thread.  In fact, maybe that's what I meant to do on line 547 which look like it does nothing.